torch/_export/serde/serialize.py

```diff
@@ -242,6 +247,22 @@ def serialize_sym_int(s: Union[int, torch.SymInt]) -> SymInt:
             f"SymInt should be either symbol or int, got `{s}` of type `{type(s)}`"
         )
 
+def serialize_sym_float(s: Union[float, torch.SymFloat]) -> SymFloat:
+    if isinstance(s, (torch.SymFloat, sympy.Symbol, float)):
+        if symbolic_shapes.is_concrete_float(s):
+            return SymFloat.create(as_float=float(s))
+        else:
+            assert isinstance(s, (torch.SymFloat, sympy.Symbol))
+            if s.node.hint is None:
+                return SymFloat.create(as_expr=SymExpr(str(s)))
```

Comment:
Ok I'm sorry, I just realized this would be an issue. We're moving away from this string serialization format, to ASTs instead, because the string format isn't roundtrippable. That work is being done for symints and symbools in https://github.com/pytorch/pytorch/pull/140084, which is close to landing.

I think this will have to switch to AST serialization instead (which isn't difficult). If you'd like I can provide the patch.

Reply:
Yes please patch it to align to your new AST PR. Are you able to edit this PR directly?